### DISCUSSION TITLE
Issues with LIS Recognizing Renamed MERRA2 Forcing Data Files (MERRA2 400 -> MERRA2 401)

### DISCUSSION THREAD

**Discussion by Hyunglok-Kim** [NASA-LIS/LISF#1501] [Q&A] [answered]
Hi,

I've noticed that since September 1, 2020, the naming convention for MERRA2 forcing data has changed from MERRA2_400.* to MERRA2_401.*. This convention had been consistent since the 2011 data sets. It appears that due to these changes, LIS is unable to correctly locate the forcing data files.

Could you please investigate this issue?

Thank you so much for your help.

Best regards,
Hyung

**Comment by karsenau** [comment]
Hi @Hyunglok-Kim:

Which version/branch of LIS are you using and have you recently updated your version in the past year?

**Reply by Hyunglok-Kim** [reply]
Hi @karsenau 
I recently complied it with LISF7.4.2 version.

**Comment by Hyunglok-Kim** [comment]
Hi @karsenau:
 have you had a chance to check this issue yet?

**Reply by karsenau** [reply]
Hi @Hyunglok-Kim,

Everyone that I've asked has had no issues with this, and a check of our code and the entire MERRA2 forcing data record has no "401" in it, that we see.

Could you please share your source of where you are getting the data from?

If you are trying to work in LDT and using the MERRA2 parameter - terrain map for processing, then yes - there is a 401 in those static files.

**Comment by Hyunglok-Kim** [comment]
Hi @karsenau 
Thank you for your prompt response. I am utilizing MERRA2 forcing data for running LIS and encountered some specificity issues that I would like to discuss.

To elaborate, through Earthdata Search (please see the file attached)
<img width="917" alt="MERRA2_401" src="https://github.com/NASA-LIS/LISF/assets/28404527/b0a39487-79b2-4af0-8931-cfd5f7d38d70">
, one can download the MERRA2 forcing data files named in the format MERRA2_x_tavg1_2d_slv_Nx.x.nc4. For instance, I generated a download link for the tavg1_2d_slv_Nx dataset, yielding the following URL from Earthdata:

https://data.gesdisc.earthdata.nasa.gov/data/MERRA2/M2T1NXSLV.5.12.4/2020/09/MERRA2_401.tavg1_2d_slv_Nx.20200930.nc4

Here, the filename for the 20200930.nc4 file includes x_401.x.nc4, differing from the expected x_400.x.nc4. However, for the 20201001 tavg1_2d_slv_Nx data, I received a correctly named link:

https://data.gesdisc.earthdata.nasa.gov/data/MERRA2/M2T1NXSLV.5.12.4/2020/10/MERRA2_400.tavg1_2d_slv_Nx.20201001.nc4

I attempted to address the discrepancy by modifying the filename from 401 to 400 for the problematic file, only to find that the download link became invalid.

This issue isn't isolated to tavg1_2d_slv_Nx but also affects tavg1_2d_(lnd), (rad), and (flx) datasets for certain months since 2020. It's not a consistent issue across all months; some files are named with 401, while others are correctly named with 400.

I'm seeking guidance on the appropriate source for downloading the accurate MERRA2 files and clarification on the circumstances under which we should use 401 instead of 400. It's worth noting that when I manually change the filename from 401 to 400, LIS processes the data without any issues.

Could you advise on how to resolve this discrepancy?

Best regards,
Hyung

**Reply by karsenau** [reply]
Thanks for that detailed reply and information on what you are seeing on the EarthData Search site and what is shown there.

I was looking at the NASA GES DISC listing for the MERRA-2 hourly files, for example:
https://disc.gsfc.nasa.gov/datasets/M2T1NXSLV_5.12.4/summary?keywords=MERRA2%20tavg1_2d_slv_Nx

Direct link to file downloads:
https://goldsmr4.gesdisc.eosdis.nasa.gov/data/MERRA2/M2T1NXSLV.5.12.4/2024/01/

e.g., file:
[MERRA2_**400**.tavg1_2d_slv_Nx.**20240101**.nc4](https://goldsmr4.gesdisc.eosdis.nasa.gov/data/MERRA2/M2T1NXSLV.5.12.4/2024/01/MERRA2_400.tavg1_2d_slv_Nx.20240101.nc4)

For the listing on the Earthdata website, you may want to email them and ask about the version number change.  I'll see what the GMAO might share and will try to report back.

Best regards

**Comment by Hyunglok-Kim** [comment]
Hi @karsenau,

Yes, I also checked goldsmr4.gesdisc and giovannai, and found that for the months of September in both 2020 and 2021, the MERRA2 data is named with x.401.x, not 400.

<img width="567" alt="image" src="https://github.com/NASA-LIS/LISF/assets/28404527/6d138a3c-18b7-4b96-97cd-918dbdb439be">

https://goldsmr4.gesdisc.eosdis.nasa.gov/data/MERRA2/M2T1NXSLV.5.12.4/2020/09/

Therefore, I am curious about how the LIS team runs LIS with MERRA2 for September in 2020 and 2021 using the current version of LISF. It seems that the current MERRA2 reading module does not recognize 401 but only works with 400 (that's my initial assumption).

So, my question is: Is it okay to use 401 instead of 400 for the forcing data? It seems to work fine for me, but I need some advice.

Thank you for your prompt response and assistance.

Sincerely,
Hyunglok

**Comment by karsenau** [comment, marked as answer]
Hi @Hyunglok-Kim:

We really appreciate you bringing this to our attention (and to the outside LIS community).

Internally, we have access to the full MERRA2 record, and for the "401" files for those specific years/month (i.e., 2020-M09 and 2021, M06-09), symbolic links are used to rename the "401" files to "400".  That was the work-around.

So you actually did run into number version difference, and I apologize for not capturing that sooner. 

LIS will only read in the "400" values, so after you download, just create symbolic links to those specific "401" files and make them have "400" in the links.

Thanks so much and cheers!

**Comment by Hyunglok-Kim** [comment]
Hi @karsenau,

Understood -- I also think creating symbolic links to 401 files is better than modifying the names of the original files. 
Since I already had a lot of involvement with the LIS team during my PhD and postdoc years in the US, I will be more than happy to contribute in the future.

Thank you so much!

Cheers!

**Reply by karsenau** [reply]
Thanks @Hyunglok-Kim!
